content/blog/2019/06/2019-06-14-performance-testing-jenkins.adoc

```diff
@@ -0,0 +1,201 @@
+---
+layout: post
+title: "Micro-benchmarking Framework for Jenkins Plugins"
+tags:
+- jmh
+- plugins
+- benchmark
+- performance
+- developer
+- gsoc
+- gsoc2019
+author: abhyudayasharma
+---
+
+= Micro-benchmarking Framework for Jenkins Plugins
+
+I have been working on improving the performance of the Role Strategy Plugin as a part of my link:https://summerofcode.withgoogle.com/projects/#6612887823450112[Google Summer of Code project].
+Since there was no existing way to measure performance and do benchmarks on Jenkins Plugins,
+my work for the first phase of this Google Summer of Code project was to create a framework for running
+benchmarks in Jenkins Plugins with a Jenkins instance available. To make our job a bit easier,
+we chose link:http://openjdk.java.net/jeps/230[Java Microbenchmark Harness] for running these benchmarks. This
+allows us to reliably measure performance of our time-critical functions and will help make Jenkins perform faster
+for everyone.
+
+The micro-benchmarking framework was recently released in the Jenkins Unit Test Harness.
```

Comment:
```suggestion
The micro-benchmarking framework was recently released in the Jenkins Unit Test Harness 2.50.
```

Maybe also makes sense to link https://github.com/jenkinsci/jenkins-test-harness/releases/tag/jenkins-test-harness-2.50